packages/frontend/app/(dashboard)/course/[cid]/insights/utils/types.ts

```diff
@@ -0,0 +1,285 @@
+import { ChartConfig } from '@/app/components/ui/chart'
+import { CurveType } from 'recharts/types/shape/Curve'
+import {
+  NameType,
+  Payload as TooltipPayload,
+  Formatter as TooltipFormatter,
+} from 'recharts/types/component/DefaultTooltipContent'
+import { Formatter } from 'recharts/types/component/DefaultLegendContent'
+import { numToWeekday } from '@koh/common'
+import {
+  generateAxisRange,
+  getMinutesToTime,
+} from '@/app/(dashboard)/course/[cid]/insights/utils/functions'
+export type ChartSize =
+  | 'xxs'
+  | 'xs'
+  | 'sm'
+  | 'md'
+  | 'lg'
+  | 'xl'
+  | '2xl'
+  | '3xl'
+  | '4xl'
+
+export const AxisChartClasses = {
+  xxs: 'mx-auto chart-xxs',
+  xs: 'mx-auto chart-xs sm:chart-xxs',
+  sm: 'mx-auto chart-sm sm:chart-xxs md:chart-xs',
+  md: 'mx-auto chart-md sm:chart-xs md:chart-sm',
+  lg: 'mx-auto chart-lg sm:chart-sm md:chart-md',
+  xl: 'mx-auto chart-xl sm:chart-md md:chart-lg',
+  '2xl': 'mx-auto chart-2xl sm:chart-md md:chart-lg',
+  '3xl': 'mx-auto chart-3xl sm:chart-md md:chart-lg',
+  '4xl': 'mx-auto chart-4xl sm:chart-md md:chart-lg',
+}
+
+export const RadialChartClasses = {
+  xxs: 'mx-auto aspect-square chart-xxs',
+  xs: 'mx-auto aspect-square chart-xs sm:chart-xxs',
+  sm: 'mx-auto aspect-square chart-sm sm:chart-xxs md:chart-xs',
+  md: 'mx-auto aspect-square chart-md sm:chart-xs md:chart-sm',
+  lg: 'mx-auto aspect-square chart-lg sm:chart-md md:chart-md',
+  xl: 'mx-auto aspect-square chart-xl sm:chart-md md:chart-lg',
+  '2xl': 'mx-auto aspect-square chart-2xl sm:chart-md md:chart-lg',
+  '3xl': 'mx-auto aspect-square chart-3xl sm:chart-md md:chart-lg',
+  '4xl': 'mx-auto aspect-square chart-4xl sm:chart-md md:chart-lg',
+}
```

Comment:
this is an interesting way to organise this!

Reply:
is this bad 🤔

Reply:
I don't know! :D

Though they are used across multiple files, so it does make sense

Reply:
i did it because while i'd love to use a `mx-auto aspect-square ${chart-size-string}`, tailwind does not support that and only does the styling for class names available at compile-time from what I've discovered